concert/base.py

```diff
@@ -66,12 +81,25 @@ def _find_in_dict(dictionary):
             # parameter in a constructor we need to bump index by one. Thus, use blacklist names
             # which won't be picked up because they are most probably used by concert or it's
             # extensions internally.
-            if instance_name and instance_name not in ['instance', 'self']:
+            if _is_name_ok(instance_name):
                 break
     finally:
         # Cleanup as python docs suggest
         del frames
 
+    if not instance_name:
+        # Try coroutines
+        tasks = asyncio.all_tasks(loop=asyncio.get_running_loop())
+        for task in tasks:
+            # Python 3.7 compatibility, later task.get_coro()
```

Comment:
I thought we are requiring >=3.8

Reply:
Strictly speaking yes, but only for tests, for the codebase I stick with 3.7